app/src/main/java/org/mozilla/fenix/share/ShareViewModel.kt

```diff
@@ -126,12 +127,11 @@ class ShareViewModel(application: Application) : AndroidViewModel(application) {
     @VisibleForTesting
     @WorkerThread
     internal fun buildDeviceList(accountManager: FxaAccountManager): List<SyncShareOption> {
-        val activeNetwork = connectivityManager?.activeNetworkInfo
         val account = accountManager.authenticatedAccount()
 
         return when {
             // No network
-            activeNetwork?.isConnected != true -> listOf(SyncShareOption.Offline)
+            connectivityManager?.isOnline() != true -> listOf(SyncShareOption.Offline)
```

Comment:
Since we use this with `NetworkCallback` it would be ideal if we could get the connectivity status from that and pass it to this function. Since we were already using the synchronous API previously we could make a new issue and merge this PR in.

Reply:
I'm a confused on this, should we create a global variable in the view model and update it inside the callback or rather make it an argument for the `buildAppsList` method?

Reply:
Make it an argument :)

Reply:
so, ive made it as an optional argument, i'm not sure if we support optional args. Do let me know if you find problems with it. 

Also, check #7180 , i was not sure what type of issue to create